### PR TITLE
mac-decklink: Fix C++ virtual function warnings

### PR DESCRIPTION
**Pull request by JohannMG**
### Description

Created another virtual function to override and moves stubbed function to the header to remove three file warnings: `...hidden overloaded virtual function 'DecklinkBase::Activate' declared here...`

### Motivation and Context

Reduce warnings, more-correct code on mac. 

### How Has This Been Tested?

MacOS Mojave 10.14.16
 - Builds on mac
 - No longer flags errors: 
```
Scanning dependencies of target mac-decklink
[ 37%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/plugin-main.cpp.o
[ 38%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-devices.cpp.o
[ 38%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-output.cpp.o
[ 38%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-source.cpp.o
[ 38%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/DecklinkOutput.cpp.o
[ 39%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/DecklinkInput.cpp.o
[ 39%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/DecklinkBase.cpp.o
[ 39%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-device-instance.cpp.o
[ 39%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-device-discovery.cpp.o
[ 40%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-device.cpp.o
[ 40%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/decklink-device-mode.cpp.o
[ 40%] Building C object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/audio-repack.c.o
[ 41%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/platform.cpp.o
[ 41%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/__/util.cpp.o
[ 41%] Building CXX object plugins/decklink/mac/CMakeFiles/mac-decklink.dir/decklink-sdk/DeckLinkAPIDispatch.cpp.o
[ 41%] Linking CXX shared module mac-decklink.so
[ 41%] Built target mac-decklink
```

### Types of changes

- Performance enhancement (non-breaking change which improves efficiency) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/decklink/DecklinkBase.cpp [hunk 1]
```diff
@@ -15,4 +15,8 @@ bool DecklinkBase::Activate(DeckLinkDevice *, long long)
 	return false;
 }
 
-void DecklinkBase::Deactivate() {}
```
Comment (e00E) [inline plugins/decklink/DecklinkBase.cpp:18]: For consistency I'd keep this in the file instead of the header like all the other functions.

**Comment by JohannMG** [comment]
@e00E 
> Alternatively you could rename the Activate version with more arguments and keep it non virtual as it's only used in DeckLinkInput.

I could. I rather both have the default implementation since that was the original intent looking at past code.

File: plugins/decklink/DecklinkBase.cpp (hunk 1 above)
Comment (JohannMG) [reply plugins/decklink/DecklinkBase.cpp:18]: The reason I made this choice is the other methods have default returns of` false`. 
This is a no-op and must be implemented by the class above. Making it clearly abstract/nop in the header can prevent this oversight to clarify that it should be overridden.

Comment (e00E) [reply plugins/decklink/DecklinkBase.cpp:18]: If you want to make it abstract (pure virtual, force others to implement) you should do `virtual void Deactivate() = 0;`
`virtual void Deactivate(){};` creates a default implementation with an empty body.
(Either way it's not a big deal, just pointing it out.)

Comment (JohannMG) [reply plugins/decklink/DecklinkBase.cpp:18]: GAH true that's what I meant to do. Might mend in a few

**Comment by JohannMG** [comment]
Changed to make `Deactivate` abstract.

**Pull request merged**